python/test/test_algo_bench.py

```diff
@@ -1,102 +1,96 @@
-    )
-    parser.add_argument(
-        "--application", default=app, choices=options[1], help="Application to run (default: %(default)s)",
-    )
-    parser.add_argument("--source-nodes", default=source_nodes, help="Source nodes file(default: %(default)s)")
-    parser.add_argument("--trials", type=int, default=trails, help="Number of trials (default: %(default)s)")
-    parser.add_argument("--num-sources", type=int, default=num_sources, help="Number of sources (default: %(default)s)")
-
-    parsed_args = parser.parse_args()
+    if not parser.threads:
+        parser.threads = int(os.cpu_count())
 
-    if not os.path.isdir(parsed_args.input_dir):
-        print(f"input directory : {parsed_args.input_dir} doesn't exist")
-        sys.exit(1)
-    if not parsed_args.threads:
-        parsed_args.threads = int(os.cpu_count())
-    print(f"Using input directory: {parsed_args.input_dir} and Threads: {parsed_args.threads}")
-
-    return parsed_args
+    return parser
 
 
 def assert_routine_output(routine_output, max_time=1_000_000):
     for subproccess in routine_output:
         assert (
             0 <= routine_output[subproccess] <= max_time
-        ), f"Invalid time for subproccess {subproccess}: took {routine_output[subproccess]} ms"
+        ), f"Invalid completion time for subproccess {subproccess}: took {routine_output[subproccess]} ms"
 
 
 def assert_types_match(ground_truth, outp):
-    for sec in ground_truth:
-        assert sec in outp, f"Output missing an element: {sec}"
-        truth_type = type(ground_truth[sec])
-        out_type = type(outp[sec])
-        assert truth_type == out_type, f"Expected types for {sec} do not match - Expected:{truth_type}, Got:{out_type}"
+    for element in ground_truth:
+        assert element in outp, f"Output missing an element: {element}"
+        truth_type = type(ground_truth[element])
+        out_type = type(outp[element])
+        assert (
+            truth_type == out_type
+        ), f"Expected types for {element} do not match - Expected:{truth_type}, Got:{out_type}"
 
     return True
 
 
-def test_single_trial_gaps():
+def get_default_args():
     arguments = {
-        "json_output": "../../../bench_statistics.json",
-        "input_dir": "../../inputs/v24/propertygraphs",
+        "json_output": os.path.join(os.path.dirname(__file__), "../../../../../../bench_statistics.json"),
+        "input_dir": os.path.join(os.path.dirname(__file__), "../../../../../inputs/v24/propertygraphs"),
         "graph": "rmat15",
         "app": "all",
         "source_nodes": "",
-        "trails": np.random.randint(1, 15),
-        "num_sources": np.random.randint(4, 64),
+        "trails": 1,
+        "num_sources": 4,
         "thread_spin": False,
         "threads": None,
     }
+    return arguments
+
+
+def test_single_trail_gaps():
+    arguments = get_default_args()
+    run_on_all_graphs(arguments)
+
 
-    options = katbench.initialize_global_vars()
+def test_multi_trail_gaps():
```

Comment:
My worry here is that we seem to focus on test completion time rather than on correctness (or I might be missing something about the tests).

I suggest we change the focus on correctness. 

How long does it take to run these tests?